### ISSUE TITLE
Make `format` configurable via the config file

### ISSUE THREAD

**Issue by jackson-theisen** [terraform-linters/tflint#1343]
The [`.tflint.hcl` configuration file](https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/config.md) doesn't currently support an option to specify the format. It should be added as an optional field within the `config` object and use the same precedence as other configs that support both a CLI flag and the config file. 

Example:

```hcl
config {
  format = "compact"
}
```

**Comment by wata727** [comment]
Can you share the use case for which you want to specify the format in the config file?
In my understanding, the format is a runtime concern and not something you want to persist in the config file. Even with linter like ESLint, formatter is not supported in the config file.
https://eslint.org/docs/user-guide/configuring/configuration-files

**Comment by jackson-theisen** [comment]
> Can you share the use case for which you want to specify the format in the config file?
> In my understanding, the format is a runtime concern and not something you want to persist in the config file. Even with linter like ESLint, formatter is not supported in the config file.
> https://eslint.org/docs/user-guide/configuring/configuration-files

The use case I had in mind would apply to a centralized CI/CD strategy that executes `tflint` across various repositories in an Organization. In this scenario, there is commonly a standard image used for each pipeline. The behavior of a pipeline would be dictated by reasonable defaults pre-configured in the image (e.g. `/home/${USER_NAME}/.tflint.hcl` is dropped into the image for the default behavior of `tflint`), and any non-default behavior could be achieved by overriding via CLI flags or a local `.tflint.hcl` file within a given repo. So if I want all of my pipelines to use the `compact` format, I have to explicitly pass `--format=compact` in every single build script. I'd prefer to set `format = "compact"` in the aforementioned default config file, then for individual jobs/repos requiring a different format, you'd pass the `--format` flag only within said repo.

**Comment by bendrucker** [comment]
An example of outputs in the config file:

https://golangci-lint.run/usage/configuration/#output-configuration

In the mean time, if you control the image, this is easily solved by a wrapper script.

**Comment by jackson-theisen** [comment]
@bendrucker I appreciate the reference and suggestion to use a wrapper script. That said, my preference would be to stay within the bounds of tflint's native configuration mechanism.

Would folks be open to the idea of an environment variable instead? Example:

```sh
TFLINT_FORMAT=compact tflint
```

The implementation could be similar to https://eslint.org/docs/developer-guide/working-with-custom-formatters#using-environment-variables and also maintains the philosophy of keeping `format` as a runtime concern (cc @wata727)

**Comment by bendrucker** [comment]
Not rejecting your suggestion just saying you could easily go ahead and roll out what you want today and then transparently remove the wrapper in the future. The proposed wrapper script would be in your `$PATH` as `tflint` and call through to the actual TFLint binary. Then in the future you push a new image version with a native config mechanism instead of the wrapper.

**Comment by bendrucker** [comment]
The link you shared refers to custom formatters using environment variables to read configuration as opposed to ESLint itself.

To me the idea behind omitting `--format` is that formatting is dependent on context. Locally I'm fine with the default, but in GitHub Actions I want a parse-able format so that annotations can be automatically added. If you set `format` in the config, users end up with a worse local experience by default.

An environment variable does address that downside. Making a special case for this isn't great though. We could borrow an idea from Terraform here:

https://www.terraform.io/cli/config/environment-variables#tf_cli_args-and-tf_cli_args_name

In this case, `TFLINT_CLI_ARGS`. In effect this would be similar to the wrapper script I suggested, but with the argument concatenation logic occurring within TFLint itself.

**Comment by wata727** [comment]
I think the use case is reasonable enough. I agree to add `format` option to the config file.
An environment variable is an implicit option, and I think it's best to use a config file or flag unless you have some particular reason.
